scripts/dedupe_examples_wasm.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+"""Deduplicate Defold example WebAssembly artifacts.
+
+This script scans the example directories, moves identical `*.wasm` and
+`*_wasm.js` files into a shared `examples/wasm/` cache keyed by the file's MD5,
+updates each example's `dmloader.js` to load from the shared cache, and ensures
+engine loader helpers point to the new locations.
+
+Run automatically from `update.py` after importing examples or manually with
+`python3 scripts/dedupe_examples_wasm.py` to shrink the published site size.
+"""
+from __future__ import annotations
+
+import hashlib
+import re
+import shutil
+import sys
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, Iterable, Tuple
+
+ROOT = Path(__file__).resolve().parent.parent
+EXAMPLES_DIR = ROOT / "examples"
+DEDUP_DIR = EXAMPLES_DIR / "wasm"
+
+
+@dataclass(frozen=True)
+class Artifact:
+    property_name: str
+    filename: str
+    suffix: str
+    size_property: str
+
+
+ARTIFACTS: Tuple[Artifact, ...] = (
+    Artifact("wasm_file", "Defoldexamples.wasm", ".wasm", "wasm_size"),
+    Artifact("wasm_pthread_file", "Defoldexamples_pthread.wasm", ".wasm", "wasm_pthread_size"),
+    Artifact("wasmjs_file", "Defoldexamples_wasm.js", ".wasm.js", "wasmjs_size"),
+    Artifact("wasmjs_pthread_file", "Defoldexamples_pthread_wasm.js", ".wasm.js", "wasmjs_pthread_size"),
+)
+
+
+def md5_digest(path: Path) -> str:
+    hasher = hashlib.md5()
+    with path.open("rb") as f:
+        for chunk in iter(lambda: f.read(1 << 20), b""):
+            hasher.update(chunk)
+    return hasher.hexdigest()
+
+
+def ensure_absolute_url(url: str) -> str:
+    if not url.startswith("/"):
+        return f"/examples/wasm/{url}"
+    return url
+
+
+def dedupe_artifact(source: Path, artifact: Artifact) -> Tuple[str, int]:
+    digest = md5_digest(source)
+    hashed_name = f"{digest}{artifact.suffix}"
+    target_path = DEDUP_DIR / hashed_name
+    if not target_path.exists():
+        target_path.parent.mkdir(parents=True, exist_ok=True)
+        shutil.copy2(source, target_path)
+    size = target_path.stat().st_size
+    # Remove the original artifact to avoid duplication.
+    try:
+        source.unlink()
+    except FileNotFoundError:
+        pass
+    return ensure_absolute_url(hashed_name), size
+
+
+def insert_property_block(content: str, values: Dict[str, str]) -> str:
+    if "wasm_file:" in content:
+        return content
+
+    def repl(match: re.Match[str]) -> str:
+        asmjs_line = match.group(0)
+        block = (
+            f"{asmjs_line}"
+            f"    wasm_file: \"{values['wasm_file']}\",\n"
+            f"    wasm_pthread_file: \"{values['wasm_pthread_file']}\",\n"
+            f"    wasmjs_file: \"{values['wasmjs_file']}\",\n"
+            f"    wasmjs_pthread_file: \"{values['wasmjs_pthread_file']}\",\n"
```

Comment:
it's far a way from perfect. It would be nice to have more robust way to do it, but it works for now